pkg/api/v1/atlasdeployment_types.go

```diff
@@ -304,10 +307,28 @@ type ProcessArgs struct {
 
 func (specArgs ProcessArgs) ToAtlas() (*mongodbatlas.ProcessArgs, error) {
 	result := &mongodbatlas.ProcessArgs{}
+	if err := convertOplogMinRetentionHours(&specArgs, result); err != nil {
+		return nil, err
+	}
+
 	err := compat.JSONCopy(result, specArgs)
 	return result, err
 }
 
+func convertOplogMinRetentionHours(specArgs *ProcessArgs, atlasArgs *mongodbatlas.ProcessArgs) error {
+	if specArgs != nil && specArgs.OplogMinRetentionHours != "" {
+		OplogMinRetentionHours, err := strconv.ParseFloat(specArgs.OplogMinRetentionHours, 64)
+		if err != nil {
+			return err
+		}
+
+		atlasArgs.OplogMinRetentionHours = &OplogMinRetentionHours
+		specArgs.OplogMinRetentionHours = ""
```

Comment:
Don't you think changing the spec value in that way may lead to side effects? Someone working in this field would not expect it was altered after calling `ToAtlas()`

Reply:
Yeah, I agree that is should fine in this case as the changes only affect `ToAtlas()` scope.